### PR TITLE
update pty dep

### PR DESCRIPTION
**Pull request by anandkumarpatel**
cleaned up a lot of broken test 
removed unused functions
sync primus and substream with API
update node

### CODE REVIEW

**Comment by bkendall** [comment]
tests failed... :(

**Comment by bkendall** [comment]
is filibuster a linux service? where does it run? I'm asking because you're upping node to `0.12` - but I'm not sure if that's fine...

**Comment by anandkumarpatel** [comment]
filibuster is a linux service, to deploy this we need to change ansible to use n to use a specific node version

**Comment by anandkumarpatel** [comment]
other option, since this is now using docker exec, is we containerize it and git it docker client

**Comment by bkendall** [comment]
I mean, if this works, lets go for it. however it needs ansible changes, so why don't you make a PR for those and then we can merge these together? (since this will break if we deploy it after we merge)

File: package.json [hunk 1]
```diff
@@ -23,17 +23,22 @@
...
     "ws": "0.4.31",
-    "substream": "0.1.1",
+    "substream": "0.1.3",
     "commander": "2.2.0",
     "dotenv": "^0.4.0"
   },
   "devDependencies": {
     "lab": "3.2.1"
+  },
+  "engineStrict": true,
```
Comment (bkendall) [inline package.json]: TIL: `engineStrict` is going to be deprecated. just a heads up - I know `pty` isn't backwards compatible, so this makes sense, but just fyi... https://docs.npmjs.com/files/package.json#enginestrict

File: package.json [hunk 2]
```diff
@@ -23,17 +23,22 @@
...
     "eson": "0.5.0",
     "primus": "2.4.0",
     "ws": "0.4.31",
-    "substream": "0.1.1",
+    "substream": "0.1.3",
     "commander": "2.2.0",
     "dotenv": "^0.4.0"
   },
   "devDependencies": {
     "lab": "3.2.1"
```
Comment (bkendall) [inline package.json]: for the future: should update `lab` :)

**Comment by bkendall** [comment]
want to make the ansbile changes (are there any now?) and get it on staging to test?

**Comment by bkendall** [comment]
LGTM

**Comment by anandkumarpatel** [comment]
no changes for ansible, already deployed on stage

**Comment by anandkumarpatel** [comment]
can verify by just going to any terminal.
http://stage-web-codenow.runnableapp.com/CodeNow/stage-charon/

**Comment by bkendall** [comment]
LGTM :+1:

**Pull request merged**